app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -211,7 +211,16 @@ def enabled_repos
 
     #api :POST, "/environments/:environment_id/consumers", N_("Register a consumer in environment")
     def consumer_create
-      host = Katello::RegistrationManager.process_registration(rhsm_params, find_content_view_environments)
+      content_view_environments = find_content_view_environments
+      if content_view_environments.any? { |cve| cve.content_view.generated_for_repository_export? }
```

Comment:
Dont think this is necessary because of  https://github.com/Katello/katello/blob/master/app/models/katello/host/content_facet.rb#L47  that checks for .generated_for_repository_export?